src/Service/ScopeHintService.php

```diff
@@ -24,12 +24,22 @@ public function __construct(
 
     public function getAllScopes() : array
     {
-        //$configArray = $this->treePaths->generate($this->scopeConfig->getValue(null));
+        $configArray = $this->scopeConfig->getValue(null);
 
-        //$all_configs = $this->scopeConfig->getValue(null);
+        $result = array();
+        $this->displayArrayRecursively($configArray, $result, "");
+        return $result;
+    }
 
+    function displayArrayRecursively($array, &$result_array, $path) : void {
 
-        return [];
+        foreach ($array as $key => $value) {
+            if (is_array($value)) {
+                $this->displayArrayRecursively($value, $result_array, $path . $key . '/');
```

Comment:
That's a good usecase for recursive function calls. It's also good to have the "recursive" keyword in the function name 👍